examples/kubernetes/node-init/node-init.yaml

```diff
@@ -45,41 +52,53 @@ spec:
               WantedBy=multi-user.target
               EOF
 
-              if [ -d "/etc/systemd/system/" ]; then
-                mv /tmp/sys-fs-bpf.mount /etc/systemd/system/
-                echo "Installed sys-fs-bpf.mount to /etc/systemd/system/"
-              elif [ -d "/lib/systemd/system/" ]; then
-                mv /tmp/sys-fs-bpf.mount /lib/systemd/system/
-                echo "Installed sys-fs-bpf.mount to /lib/systemd/system/"
-              fi
-
-              systemctl enable sys-fs-bpf.mount
-              systemctl start sys-fs-bpf.mount
+                if [ -d "/etc/systemd/system/" ]; then
+                  mv /tmp/sys-fs-bpf.mount /etc/systemd/system/
+                  echo "Installed sys-fs-bpf.mount to /etc/systemd/system/"
+                elif [ -d "/lib/systemd/system/" ]; then
+                  mv /tmp/sys-fs-bpf.mount /lib/systemd/system/
+                  echo "Installed sys-fs-bpf.mount to /lib/systemd/system/"
+                fi
 
-              echo "Changing kubelet configuration to --network-plugin=cni --cni-bin-dir=/home/kubernetes/bin"
-              mkdir -p /home/kubernetes/bin
-              sed -i "s:--network-plugin=kubenet:--network-plugin=cni\ --cni-bin-dir=/home/kubernetes/bin:g" /etc/default/kubelet
-              echo "Restarting kubelet..."
-              systemctl restart kubelet
+                systemctl enable sys-fs-bpf.mount
+                systemctl start sys-fs-bpf.mount
 
-              echo "Installing /etc/cni/net.d/04-cilium-cni.conf"
-              mkdir -p /etc/cni/net.d/
-              cat >/etc/cni/net.d/04-cilium-cni.conf <<EOF
+                echo "Installing /etc/cni/net.d/04-cilium-cni.conf"
+                mkdir -p /etc/cni/net.d/
+                cat >/etc/cni/net.d/04-cilium-cni.conf <<EOF
               {
-                  "name": "cilium",
-                  "type": "cilium-cni"
+                "name": "cilium",
+                "type": "cilium-cni"
               }
               EOF
 
-              if ip link show cbr0; then
-                      echo "Detected cbr0 bridge. Deleting interface..."
-                      ip link del cbr0
-              fi
+                echo "Changing kubelet configuration to --network-plugin=cni --cni-bin-dir=/home/kubernetes/bin"
+                mkdir -p /home/kubernetes/bin
+                sed -i "s:--network-plugin=kubenet:--network-plugin=cni\ --cni-bin-dir=/home/kubernetes/bin:g" /etc/default/kubelet
+                echo "Restarting kubelet..."
+                systemctl restart kubelet
+
+                echo "Restarting kubenet managed pods"
+                if grep -q 'docker' /etc/crictl.yaml; then
+                  # Works for COS, ubuntu
+                  for f in `find /var/lib/cni/networks/ -type f ! -name lock ! -name last_reserved_ip.0`; do docker rm -f $(cat $f) || true; done
```

Comment:
This seems very aggressive. Any reason not to do this via the apiserver? At the very least I would hide this behind an option. This could have a lot of consequences that we don't want.

Reply:
>At the very least I would hide this behind an option.

Yeah I was thinking the same.

>Any reason not to do this via the apiserver?

The user will have to remember to do that step in the kube-apiesrver every time a new node is provisioned.